src/test/java/com/amihaiemil/zold/RtWalletTestCase.java

```diff
@@ -0,0 +1,78 @@
+package com.amihaiemil.zold;
+
+import com.amihaiemil.zold.mock.AssertRequest;
+import com.amihaiemil.zold.mock.Condition;
+import com.amihaiemil.zold.mock.MockHttpClient;
+import com.amihaiemil.zold.mock.Response;
+import org.apache.hc.client5.http.classic.HttpClient;
+import org.apache.hc.core5.http.ClassicHttpResponse;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpStatus;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Provides a test case for the {@link RtWallet} operations.
+ */
+public class RtWalletTestCase {
+
+    /**
+     * A base URI for testing.
+     */
+    private static final String BASE_URI = "http://wts.test.com";
+
+    /**
+     * Checks whether the wallet can be instantiated.
+     */
+    @Test
+    public void instantiates() {
+        MatcherAssert.assertThat(
+            new RtWallet(
+                new MockHttpClient(),
+                URI.create(BASE_URI)
+            ),
+            Matchers.instanceOf(RtWallet.class)
+        );
+    }
+
+    /**
+     * Checks whether the {@link RtWallet#getId()} invocation succeeds.
+     * @throws Exception If an exception is thrown
+     */
+    @Test
+    public void getsId() throws Exception {
+        final ClassicHttpResponse response = new Response(
+            HttpStatus.SC_OK,
+            ContentType.TEXT_PLAIN,
+            "12345"
+        );
+        final HttpClient httpClient = new MockHttpClient(
+            new AssertRequest(
+                response,
+                new Condition(
+                    "",
+                    // @checkstyle LineLength (1 line)
+                    r -> {
+                        try {
+                            return (BASE_URI + "/id")
+                                .equals(r.getUri().toString());
+                        } catch (final URISyntaxException ex) {
+                            throw new RuntimeException(ex);
+                        }
+                    }
+                )
+            )
+        );
+        MatcherAssert.assertThat(
```

Comment:
@g4s8 , this is not needed, as when another assertion fails before the mentioned one, there would be only that assertion message available, e.g.:
```
java.lang.AssertionError: 
	at org.junit.Assert.fail(Assert.java:88)
	at com.amihaiemil.zold.mock.Condition.test(Condition.java:69)
	at com.amihaiemil.zold.mock.MockHttpClient.lambda$check$1(MockHttpClient.java:151)
	at java.util.Arrays$ArrayList.forEach(Arrays.java:3880)
	at com.amihaiemil.zold.mock.MockHttpClient.check(MockHttpClient.java:150)
	at com.amihaiemil.zold.mock.MockHttpClient.execute(MockHttpClient.java:105)
	at com.amihaiemil.zold.RtWallet.getId(RtWallet.java:79)
	at com.amihaiemil.zold.RtWalletTestCase.getsId(RtWalletTestCase.java:74)
```

Reply:
@denis-konovalyenko when these tests runs on CI machine, the error output will contain an error like:
> Test RtWalletTestCase.testMethodName failed

so it will be no clear for developer which assertion fails, so he/she need to understand stacktraces. When you use assertion message, this message will be displayed in console and developer can understand what is wrong in one view.

Reply:
@g4s8 , corrected. Could you please take another look?